calico/felix/config.py

```diff
@@ -374,6 +380,38 @@ def _validate_cfg(self, final=True):
                 self.parameters["DefaultEndpointToHostAction"]
             )
 
+        # Reporting interval and TTL need to be integers.
+        try:
+            self.REPORTING_TTL_SECS = int(self.REPORTING_TTL_SECS)
+        except ValueError:
+            raise ConfigException("Reporting interval was not integer.",
+                                  self.parameters["ReportingIntervalSecs"])
+        try:
+            self.REPORTING_INTERVAL_SECS = int(self.REPORTING_INTERVAL_SECS)
+        except ValueError:
+            raise ConfigException("Reporting TTL was not integer.",
+                                  self.parameters["ReportingTTLSecs"])
+
+        # For non-positive time values of reporting interval we set both
+        # interval and ttl set to 0 - i.e. status reporting is disabled.
+        if self.REPORTING_INTERVAL_SECS <= 0:
+            self.REPORTING_TTL_SECS = 0
+            self.REPORTING_INTERVAL_SECS = 0
+
+        # Status report time to live must be more than status reporting interval.
+        if self.REPORTING_TTL_SECS <= self.REPORTING_INTERVAL_SECS\
+                and self.REPORTING_TTL_SECS != 0:
```

Comment:
the reasoning behind the 'and ttl != 0' part wasn't immediately clear until I carried on reading below - if you move the setting ttl to default value if not set (or 0) to above this test then you can remove the 'and' here